WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt

```diff
@@ -21,9 +25,42 @@ class PostConflictResolutionFeatureUtils @Inject constructor(
      * the remote version.
      */
     fun getRemotePostPayloadForPush(payload: RemotePostPayload): RemotePostPayload {
-        if (isPostConflictResolutionEnabled().not()) {
+        if (isPostConflictResolutionEnabled()) {
+            setLastModifiedForConflictResolution(payload)
+        } else {
             payload.shouldSkipConflictResolutionCheck = true
         }
         return payload
     }
+
+    /**
+     * Resolves post-conflict issues for scheduled posts by adjusting the `if_not_modified_since` field. Normally, when
+     * a post is scheduled for the future, both the `dateCreated` and `lastModified` fields are set to the future
+     * publication date, as returned by the backend. This setup can prevent effective post-conflict resolution.
+     * This function introduces an additional field in the payload that is used to override the `lastModified` value.
+     * By setting a past date in `if_not_modified_since`, it ensures that the conflict resolution mechanism can operate
+     * correctly.
+     */
+    private fun setLastModifiedForConflictResolution(payload: RemotePostPayload) {
+        payload.post?.let { post ->
+            payload.lastModifiedForConflictResolution = if (shouldUpdateLastModifiedForConflictResolution(post)) {
+                dateTimeUtilsWrapper.dateStringFromIso8601MinusMillis(
+                    dateTimeUtilsWrapper.currentTimeInIso8601(),
```

Comment:
@zwarm could you please check this line? Your commit didn't work for me so I had to pass the current date minus something.

Reply:
@pantstamp very odd indeed, but I tested with current time and all is working as expected